src/index.js

```diff
@@ -26,115 +25,149 @@ const WARN = 'warning';
 const ERROR = 'error';
 
 const getBaseComponent = (components, component) => {
-  if (component in components) {
-    return components[component];
-  } else throw new Error(`No component defined with name ${component}`)
+	if (component in components) {
+		return components[component];
+	} else {
+		const error = new Error(`No component defined with name ${component}`);
+		error.internal = true;
+		throw error;
+	}
 };
 
 const renderReact = (component, data) => {
-  const rootElemComponent = React.createElement(component, data);
-  return ReactDOMServer.renderToStaticMarkup(rootElemComponent);
+	const rootElemComponent = React.createElement(component, data);
+	return ReactDOMServer.renderToStaticMarkup(rootElemComponent);
 };
 
 const html2text = (html) => {
-  return striptags(html.replace(/<br\s*\/?>/g, "\n"))
-      .split("\n")
-      .map(l => l.trim())
-      .join("\n");
+	return striptags(html.replace(/<br\s*\/?>/g, "\n"))
+		.split("\n")
+		.map(l => l.trim())
+		.join("\n");
 };
 
 const defaultLogger = (level, message) => console.log(`${new Date()} ${level}: ${message}`);
 
+const getOptions = options => {
```

Comment:
This function seems a bit too convoluted for its actual functionality, which is not far removed from the following:

```javascript
return {
  log: options.logger || defaultLogger,
  htmlFormat: input => options.minificationOptions ? minify(input, options.minificationOptions) : pretty(input, {ocd : true})
  mjmlStrict: mjmlStrict || false
};
```

Reply:
Also, it may be better to simply move it back into `createRenderServer`, unless I'm missing some other use of this function.

Reply:
I extracted it to keep the separate methods smaller